pkg/operator/controller/ingress/controller.go

```diff
@@ -493,18 +506,30 @@ func setDefaultPublishingStrategy(ic *operatorv1.IngressController, platformStat
 					if statusLB.ProviderParameters.AWS == nil {
 						statusLB.ProviderParameters.AWS = &operatorv1.AWSLoadBalancerParameters{}
 					}
-					if len(statusLB.ProviderParameters.AWS.Type) == 0 {
+					if len(statusLB.ProviderParameters.AWS.Type) == 0 || specLB.ProviderParameters.AWS.Type == "Classic" {
 						statusLB.ProviderParameters.AWS.Type = operatorv1.AWSClassicLoadBalancer
 					}
 					if statusLB.ProviderParameters.AWS.ClassicLoadBalancerParameters == nil {
 						statusLB.ProviderParameters.AWS.ClassicLoadBalancerParameters = &operatorv1.AWSClassicLoadBalancerParameters{}
 					}
-					var v metav1.Duration
-					if specIdleTimeout.Duration > 0 {
-						v = specIdleTimeout
+					// The only provider parameter that is supported
+					// for AWS is the connection idle timeout for
+					// classic ELBs.
+					var specIdleTimeout, statusIdleTimeout metav1.Duration
+					if specLB.ProviderParameters != nil && specLB.ProviderParameters.AWS != nil && specLB.ProviderParameters.AWS.ClassicLoadBalancerParameters != nil {
+						specIdleTimeout = specLB.ProviderParameters.AWS.ClassicLoadBalancerParameters.ConnectionIdleTimeout
+					}
+					if statusLB.ProviderParameters != nil && statusLB.ProviderParameters.AWS != nil && statusLB.ProviderParameters.AWS.ClassicLoadBalancerParameters != nil {
+						statusIdleTimeout = statusLB.ProviderParameters.AWS.ClassicLoadBalancerParameters.ConnectionIdleTimeout
+					}
+					if specIdleTimeout != statusIdleTimeout {
```

Comment:
Consider initializing statusLB.ProviderParameters.AWS.ClassicLoadBalancerParameters.ConnectionIdleTimeout to nil and then setting it to specIdleTimeout in this if statement. I think that might be more explicit and readable

```suggestion
if specIdleTimeout != statusIdleTimeout {
  statusLB.ProviderParameters.AWS.ClassicLoadBalancerParameters.ConnectionIdleTimeout = nil
  if specIdleTimeout.Duration > 0 {
    statusLB.ProviderParameters.AWS.ClassicLoadBalancerParameters.ConnectionIdleTimeout = specIdleTimeout
  }
  changed = true
}
```

Reply:
Actually this part of the code is not related to this PR so did not touch the logic. Nil value was not set in the earlier code as well. I think may be a separate PR can be filed for this to refactor. However I feel setting nil is not needed.

Reply:
Oh yea now I see the diff. Sorry about that, this can stay as is.